### PR TITLE
Ability for service recipe to expose known certificate hosts

### PR DESCRIPTION
**Pull request by vraravam**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has *stopped working* or is *working incorrectly*, please log the bug [here](https://github.com/getferdi/recipes/issues)
2. If you are requesting support for a **new service** in Ferdi, please log it [here](https://github.com/getferdi/recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/getferdi/ferdi#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdi, but to get new recipes between Ferdi releases, this documentation is quite useful.)
4. Please consider supporting Ferdi!
  👉  https://github.com/sponsors/getferdi
  👉  https://opencollective.com/getferdi/donate
5. Please ensure you've completed all of the following.
- [x] I have read the [Contributing Guidelines](https://github.com/getferdi/ferdi/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/getferdi/ferdi/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/getferdi/ferdi/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
This PR exposes the ability for a service recipe to implement a method `knownCertificateHosts` in its `index.js` file - and then use this to validate certificates while checking requests.

#### Motivation and Context
This should hopefully fix the issue about [ms-teams login issue](https://docs.microsoft.com/en-us/microsoftteams/troubleshoot/teams-sign-in/sign-in-loop#resolution).

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes
Ability for service recipe to expose known certificate hosts for validating request certs

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -488,6 +488,24 @@ ipcMain.on(
...
+ipcMain.on(
+  'knownCertificateHosts',
+  (e, { knownHosts, serviceId }) => {
+    debug('Received knownCertificateHosts', knownHosts, 'for serviceId', serviceId);
+    // https://www.electronjs.org/docs/api/session#sessetcertificateverifyprocproc
+    session
+      .fromPartition(`persist:service-${serviceId}`)
+      .setCertificateVerifyProc((request, callback) => {
+        const { hostname } = request;
+        if (knownHosts.find(item => item.indexOf(hostname) !== -1).length > 0) {
```
Comment (mhatvan) [inline src/index.js:500]: `item.includes(hostname)` for a more readable, modern alternative.

File: src/index.js [hunk 2]
```diff
@@ -488,6 +488,24 @@ ipcMain.on(
...
+          callback(0);
+        } else {
+          callback(-2);
```
Comment (mhatvan) [inline src/index.js:503]: Add some comments ove these `callback()` to let other devs know what this is doing.

File: src/index.js [hunk 3]
```diff
@@ -470,7 +470,7 @@ ipcMain.on('open-browser-window', (e, { url, serviceId }) => {
 ipcMain.on(
   'modifyRequestHeaders',
   (e, { modifiedRequestHeaders, serviceId }) => {
-    debug('Received modifyRequestHeaders', modifiedRequestHeaders, serviceId);
+    debug('Received modifyRequestHeaders', modifiedRequestHeaders, 'for serviceId', serviceId);
```
Comment (mhatvan) [inline src/index.js:473]: Hard to read, wrap it with template literals to be more readable.

File: src/index.js [hunk 4]
```diff
@@ -488,6 +488,24 @@ ipcMain.on(
   },
 );
 
+ipcMain.on(
+  'knownCertificateHosts',
+  (e, { knownHosts, serviceId }) => {
+    debug('Received knownCertificateHosts', knownHosts, 'for serviceId', serviceId);
```
Comment (mhatvan) [inline src/index.js:494]: Same here.

File: src/index.js (hunk 1 above)
Comment (vraravam) [reply src/index.js]: done.

File: src/index.js (hunk 2 above)
Comment (vraravam) [reply src/index.js:503]: done

File: src/index.js (hunk 3 above)
Comment (vraravam) [reply src/index.js]: done

File: src/index.js (hunk 4 above)
Comment (vraravam) [reply src/index.js]: done

**Review by mhatvan — APPROVED** [review]

**Pull request merged**